pkg/kubelet/prober/worker.go

```diff
@@ -222,6 +222,14 @@ func (w *worker) doProbe() (keepGoing bool) {
 			w.pod.Spec.RestartPolicy != v1.RestartPolicyNever
 	}
 
+	// Graceful shutdown of the pod.
+	if w.pod.ObjectMeta.DeletionTimestamp != nil && w.probeType == liveness {
```

Comment:
I'd suggest to stop startup probing on termination. Yes, it may be very rare occurrence, when startup takes so long that operator decided to terminate the pod. But if the same confusion - marking pod failed will occur in this case, it will be unlikely.

Reply:
In that case (terminated during the startup), I would suggest to output a warning log message, not verbose. As this is quite unusual situation to get into.